### PR TITLE
[XLA:CPU] [oneDNN] Layer Norm XLA HLO Pattern Matcher with oneDNN custom call rewrite

### PR DESCRIPTION
**Pull request by mahmoud-abuzaina**
This PR enables oneDNN library call for the matched XLA HLO Layer Norm pattern through custom_call instruction. In particular, this PR adds:

1. oneDNN ops rewriter pass that will rewrite custom oneDNN ops like Layer Norm and Softmax. Currently, this pass is located before the BF16 all-reduce to F32 promotion.
2. Layer Norm pattern matcher as seen in some Flax models
3. oneDNN Layer Norm custom call rewrite
4. Variable number of arguments for oneDNN custom call targets
5. A test for XLA HLO Layer Norm pattern match and rewrite verification

### CODE REVIEW

**Review by d0k — APPROVED** [review]

**Review by penpornk — APPROVED** [review]
Approving now since @d0k has already approved the original PR: https://github.com/tensorflow/tensorflow/pull/61565

File: xla/tests/onednn_layer_norm_test.cc [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+#include "xla/shape_util.h"
+#include "xla/test.h"
+#include "xla/test_helpers.h"
+#include "xla/tests/hlo_test_base.h"
+#include "xla/tests/test_macros.h"
+
+namespace xla {
+namespace {
+
+class LayerNormTest : public HloTestBase {};
```
[the remark this answers is not in the document]
Comment (penpornk) [reply xla/tests/onednn_layer_norm_test.cc:28]: Thank you, @thomasjoerg !
@mahmoud-abuzaina I've made the fix internally. The changes went in and should reflect in Github soon. You don't need to push a new commit.
